server/routes/genesetEnrichment.ts

```diff
@@ -0,0 +1,122 @@
+import { genesetEnrichmentRequest, genesetEnrichmentResponse } from '../shared/types/routes/genesetEnrichment.ts'
+import fs from 'fs'
+import { spawn } from 'child_process'
+import { Readable } from 'stream'
+import path from 'path'
+import serverconfig from '../src/serverconfig.js'
+
+export const api = {
+	endpoint: 'genesetEnrichment',
+	methods: {
+		all: {
+			init,
+			request: {
+				typeId: 'genesetEnrichmentRequest'
+			},
+			response: {
+				typeId: 'genesetEnrichmentResponse'
+				// will combine this with type checker
+				//valid: (t) => {}
+			}
+		}
+	}
+}
+
+function init({ genomes }) {
+	return async (req: any, res: any): Promise<void> => {
+		try {
+			const results = await run_genesetEnrichment_analysis(req.query as genesetEnrichmentRequest, genomes)
+			res.send(results as genesetEnrichmentResponse)
+		} catch (e: any) {
+			res.send({ status: 'error', error: e.message || e })
+		}
+	}
+}
+
+async function run_genesetEnrichment_analysis(q: genesetEnrichmentRequest, genomes: any) {
+	if (!genomes[q.genome].termdbs) throw 'termdb database is not available for ' + q.genome
+
+	const genesetenrichment_input = {
+		genes: q.genes,
+		fold_change: q.fold_change,
+		db: genomes[q.genome].termdbs.msigdb.cohort.db.connection.name,
+		gene_set_group: q.geneSetGroup
+	}
+
+	//console.log('__dirname:',__dirname)
+	//console.log('genesetenrichment_input:', JSON.stringify(genesetenrichment_input))
+
+	//fs.writeFile('test.txt', '/' + JSON.stringify(genesetenrichment_input), function (err) {
+	//	// For catching input to rust pipeline, in case of an error
+	//	if (err) return console.log(err)
+	//})
+
+	const gsea_output: any = await run_gsea(
+		path.join(serverconfig.binpath, '../python/src', 'gsea.py'),
+		'/' + JSON.stringify(genesetenrichment_input) // "/" is needed for python to accept the bracket "{" as a bracket
+	)
+
+	let result
+	for (const line of gsea_output.split('\n')) {
+		if (line.startsWith('result: ')) {
+			result = JSON.parse(line.replace('result: ', ''))
+		} else {
+			console.log(line)
+		}
+	}
+	//console.log('result:', result)
+	return result as genesetEnrichmentResponse
+}
+
+async function run_gsea(path, data) {
+	try {
+		await fs.promises.stat(path)
+	} catch (e) {
+		throw `${path} does not exist`
+	}
+	return new Promise((resolve, reject) => {
+		const _stdout: any[] = []
+		const _stderr: any[] = []
+		// spawn python process
+		let python_path = 'python3'
+		if (serverconfig.py_path) python_path = serverconfig.py_path
```

Comment:
The default python binary path should be set in `src/serverconfig.js`, like other tools. It's centralized there to minimize confusing overrides that occur randomly in the server codebase. And prefer to name this as `serverconfig.python` to be consistent with name casing of the other tools (they use camelCase instead of underscore/snake_case.)

Reply:
Changed `serverconfig.py_path` to `serverconfig.python`.